### ISSUE TITLE
document global renderlet

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#910]
I just noticed this thing existed, it is a single callback directly on `dc` which gets called directly after any  `redrawAll` or `renderAll`.

https://github.com/dc-js/dc.js/blob/develop/src/core.js#L283-L289

It fires immediately, like what we're calling pretransition #806, and unlike the other renderlet events. It takes the group name as its only parameter.

It does not use an event dispatcher. If I were to modernize it, I would suggest an event dispatcher per chart group, and (also/instead) taking the list of charts in the group. I don't know if people use it though, since it has never been documented, although it goes back almost to the beginning of the library.

**Comment by mtraynham** [comment]
I don't know if this is generally helpful, even if it does fire at the "pretransition" step.  dc.js offers very few global properties to be altered at these stages, so what does this gain that adding a listener to the chart does not?

Also I'd like to point out, that a group can contain any type of chart.  If a user is to alter the charts in some fashion during this call, it would require them to know what kind of chart it is (i.e. #904)

**Comment by gordonwoodhull** [comment]
Yeah. I bet this is really just for detecting redraws / renders,  if you're doing something else on the screen like updating some text or something. I'd like to deprecate it in favor of some chart group events. 

Wish we could find out how people are using it. Oh, we can! 

https://github.com/search?q=dc.renderlet&type=Code&utf8=✓

**Comment by mtraynham** [comment]
Interesting never tried that URL before.  Seems to just return a bunch of clones checked into source control.

**Comment by gordonwoodhull** [comment]
Ah, you are right, sorry for the noise. I thought I saw some real results in there, but nothing in the first ten pages. (This the search function at the top of github, if you press backspace then it starts searching over all repos.)

Google says `dc.renderlet` was implemented in response to #20, for updating the document after interactions. So yeah, chart group events should cover it. And you're right, it has nothing to do with rendering (or even the charts) so it shouldn't be called renderlet.

**Comment by r4j4h** [comment]
The only use I'd see for it you've mentioned and would be provided for by chart (group) events so +1 for deprecating it in favor of cleaner, more semantically explicit events! :)

**Comment by gordonwoodhull** [comment]
For now, minimal documentation as I just posted on gitter:

Usage: 
```js
dc.renderlet(function() {
     // do what you need here
})
```
Replaces any previous callback.

I'll add this to the docs on my next sweep for 2.0. The interface won't change in 2.0.* but in 2.1 we should do better named event dispatches per chart group.
